### DISCUSSION TITLE
Sicurezza: connessioni da cloud a home network

### DISCUSSION THREAD

**Discussion by kronos72it** [indomus/forum#2724] [Networking]
Vorrei iniziare a usare NS Panel Pro ed ho letto la recenzione [qui](https://indomus.it/componenti/recensione-itead-sonoff-nspanel-pro-il-display-touch-che-funge-da-hub-e-bridge-zigbee/) .   Cmq a prescindere dal tipo di sistema che si vuole realizzare ho compreso che questi si appoggiano ad un cloud del costruttore . QUindi i miei dubbi sono : 
1. come fa il cloud a fare delle chiamate entranti verso la mia rete locale tenendo conto che sul router tutte le porte sono chiuse ( guai se cosi non fosse 
2. nel caso di NSPanel pro si può evitare di appoggiarsi al cloud di ITEAD e controllare il tutto con HomeAssistant ?

**Comment by KC-inDomus** [comment]
> Vorrei iniziare a usare NS Panel Pro ed ho letto la recenzione [qui](https://indomus.it/componenti/recensione-itead-sonoff-nspanel-pro-il-display-touch-che-funge-da-hub-e-bridge-zigbee/) . Cmq a prescindere dal tipo di sistema che si vuole realizzare ho compreso che questi si appoggiano ad un cloud del costruttore . QUindi i miei dubbi sono :
> 
> 1. come fa il cloud a fare delle chiamate entranti verso la mia rete locale tenendo conto che sul router tutte le porte sono chiuse ( guai se cosi non fosse
> 2. nel caso di NSPanel pro si può evitare di appoggiarsi al cloud di ITEAD e controllare il tutto con HomeAssistant ?

1 questo vale per qualsiasi host presente sulla LAN. è ovvio che possa parlare col cloud: l'host apre una porta alta random e il nat del router fa sì che il cloud possa parlargli
2 sì. ti basta seguire la nostra guida e installare l'app home assistant companion.

**Reply by kronos72it** [reply]
Parlando ad es. del citofono WIFi: 
1. quindi è il citofono interno alla rete che apre la connessione verso cloud (esterno) ?  se cosi fosse allora quale è la dinamica quando io da app apro il cancello ?  host fa polling verso il cloud ? 
Il caso in esame è relativo ad una chiamata asincrona dal mio cell (app) -->cloud-->citofono (host intenro)  , mase cosi fosse  essendo la richiesta originata dall esterna non potrebbe raggiungere host.

grazieee

**Reply by snakuzzo** [reply]
Host sulla LAN apre connessione col Cloud
L'app del cell si collega al Cloud
Sul router in entrata non viene aperta alcuna porta di ascolto.

Dall'app fai richiesta verso il servizio cloud, che veicola poi la richiesta verso l'host sulla tua LAN sfruttando il tunnel creato appunto dallo stesso host.

**Reply by kronos72it** [reply]
quindi host usa una richiesta KeepAlive o simile ?
